src/parser/Statement.ts

```diff
@@ -1274,11 +1284,53 @@ export class InterfaceStatement extends Statement implements TypedefProvider {
         }
     }
 
+    public buildSymbolTable(parentIface?: Statement) {
+        this.memberTable.clear();
+        if (isInterfaceStatement(parentIface)) {
+            this.memberTable.setParent(parentIface?.memberTable);
+        }
+
+        for (const statement of this.methods) {
+            const funcType = statement?.func.getFunctionType();
+            this.memberTable.addSymbol(statement?.name?.text, statement?.range, funcType);
+        }
+        for (const statement of this.fields) {
+            this.memberTable.addSymbol(statement?.tokens.name?.text, statement?.range, statement.getType());
+        }
+    }
+
+    public hasParent() {
+        return !!this.parentInterfaceName;
+    }
+
+    public getParentName() {
+        return !!this.parentInterfaceName;
+    }
+
     /**
-     * The name of the interface (without the namespace prefix)
+     * Gets an array of possible parent class names, taking into account the namespace this class was created under
+     * @returns array of possible parent class names
      */
-    public get name() {
-        return this.tokens.name?.text;
+    public getPossibleFullParentNames(): string[] {
+        if (!this.hasParent()) {
+            return [];
+        }
+        if (this.parentInterfaceName?.getNameParts().length > 1) {
+            // The specified parent class already has a dot, so it must already reference a namespace
+            return [this.parentInterfaceName.getName()];
+        }
+        const names = [];
+
+        if (this.namespaceName) {
+            // We're under a namespace, so the full parent name MIGHT be with this namespace too
+            names.push(this.namespaceName.getName() + '.' + this.parentInterfaceName.getName());
+        }
+        names.push(this.parentInterfaceName.getName());
+        return names;
+    }
+
+    public getThisBscType(): InterfaceType {
```

Comment:
Could we set the type of `m` and then just call something like `getSymbol("m").getBscType()`?

Reply:
`getThisBscType()` is more about getting the resulting type of a statement
I'm open to renaming it, but this coupled with the `TypeExpression` stuff (and maybe refactored a little) will be pretty powerful... you could (for many statements) get the type directly... especially if it is declared explicitly (e.g. ` blah as string`, etc) ...